scripts/src/pullrequest/prepare_pr_comment.py

```diff
@@ -38,138 +43,245 @@ def get_look_at_job_output_comment():
-    msg = "\n---\n\n"
-    msg += "For information on the certification process see:\n"
-    msg += "- [Red Hat certification requirements and process for Kubernetes applications that are deployed using Helm charts.](https://redhat-connect.gitbook.io/partner-guide-for-red-hat-openshift-and-container/helm-chart-certification/overview).\n\n"
-    msg += "For support, connect with our [Technology Partner Success Desk](https://redhat-connect.gitbook.io/red-hat-partner-connect-general-guide/managing-your-account/getting-help/technology-partner-success-desk).\n\n"
-    msg += f'/metadata {{"vendor_label": "{vendor_label}", "chart_name": "{chart_name}"}}\n\n'
+def metadata_label(vendor_label, chart_name):
+    """Returns the metadata context that must suffix PR comments."""
+    return (
+        f'/metadata {{"vendor_label": "{vendor_label}", "chart_name": "{chart_name}"}}'
+    )
+
+
+def task_table(task_tuples):
+    """returns a markdown table containing tasks and their outcome
+
+    Args:
+        task_tuples: a list of two-length tuples where index 0 is the task
+          and index 1 is the outcome. These values should be short.
+    """
+    sorted(task_tuples)
+    msg = "|task|outcome|" + "\n|---|---|"
+    for task_tuple in task_tuples:
+        msg += f"\n|{task_tuple[0]}|{task_tuple[1]}|"
     return msg
 
 
+def overall_outcome(outcome):
+    return append_to("### Outcome:", f"**{outcome}**")
+
+
 def main():
     pr_content_result = sys.argv[1]
     run_verifier_result = sys.argv[2]
     verify_result = sys.argv[3]
     issue_number = open("./pr/NR").read().strip()
     vendor_label = open("./pr/vendor").read().strip()
     chart_name = open("./pr/chart").read().strip()
-    msg = get_comment_header(issue_number)
+
+    community_manual_review = os.environ.get("COMMUNITY_MANUAL_REVIEW", False)
     oc_install_result = os.environ.get("OC_INSTALL_RESULT", False)
 
+    msg = f"Thank you for submitting PR #{issue_number} for Helm Chart Certification!"
+
+    # Assemble the detail separately to control order in which it is added to
+    # the overall output.
+    detail_message = "### Detail"
+
+    outcome = "Failed"
+
     # Handle success explicitly
     if (
         pr_content_result == "success"
         and run_verifier_result == "success"
         and verify_result == "success"
         and oc_install_result == "success"
     ):
-        msg += prepare_success_comment()
+        outcome = "Passed"
+        detail_message = append_to(detail_message, get_success_coment())
         gitutils.add_output("pr_passed", "true")
     else:  # Handle various failure scenarios.
         if pr_content_result == "failure":
-            msg += prepare_pr_content_failure_comment()
+            detail_message = append_to(
+                detail_message, prepare_pr_content_failure_comment()
+            )
             gitutils.add_output("pr_passed", "false")
         elif run_verifier_result == "failure":
```

Comment:
We should consider displaying error messages for each failure that was received rather than only for the first one.

Reply:
Definitely worth considering, but I think I'd like to defer this to a future PR.

Reply:
Opened https://github.com/openshift-helm-charts/development/issues/286 to track that improvement